tsp-output/@azure-tools/typespec-csharp/src/Generated/Models/CreateTranslationRequest.Serialization.cs

```diff
@@ -2,180 +2,57 @@
-                }
-                if (property.NameEquals("model"u8))
-                {
-                    model = new CreateTranslationRequestModel(property.Value.GetString());
-                    continue;
-                }
-                if (property.NameEquals("prompt"u8))
-                {
-                    prompt = property.Value.GetString();
-                    continue;
-                }
-                if (property.NameEquals("response_format"u8))
-                {
-                    if (property.Value.ValueKind == JsonValueKind.Null)
-                    {
-                        continue;
-                    }
-                    responseFormat = new CreateTranslationRequestResponseFormat(property.Value.GetString());
-                    continue;
-                }
-                if (property.NameEquals("temperature"u8))
-                {
-                    if (property.Value.ValueKind == JsonValueKind.Null)
-                    {
-                        continue;
-                    }
-                    temperature = property.Value.GetDouble();
-                    continue;
-                }
-                if (options.Format != "W")
-                {
-                    additionalPropertiesDictionary.Add(property.Name, BinaryData.FromString(property.Value.GetRawText()));
-                }
+                // TODO: preferred way to handle floats/numerics?
+                content.Add(new StringContent($"{Temperature}"), "temperature");
             }
-            serializedAdditionalRawData = additionalPropertiesDictionary;
-            return new CreateTranslationRequest(file, model, prompt.Value, OptionalProperty.ToNullable(responseFormat), OptionalProperty.ToNullable(temperature), serializedAdditionalRawData);
-        }
 
-        BinaryData IPersistableModel<CreateTranslationRequest>.Write(ModelReaderWriterOptions options)
-        {
-            var format = options.Format == "W" ? ((IPersistableModel<CreateTranslationRequest>)this).GetFormatFromOptions(options) : options.Format;
-
-            switch (format)
+            string contentType = default;
+            if (content.Headers.ContentType is MediaTypeHeaderValue contentTypeValue)
             {
-                case "J":
-                    return ModelReaderWriter.Write(this, options);
-                default:
-                    throw new FormatException($"The model {nameof(CreateTranslationRequest)} does not support '{options.Format}' format.");
+                contentType = contentTypeValue.ToString();
             }
-        }
 
-        CreateTranslationRequest IPersistableModel<CreateTranslationRequest>.Create(BinaryData data, ModelReaderWriterOptions options)
-        {
-            var format = options.Format == "W" ? ((IPersistableModel<CreateTranslationRequest>)this).GetFormatFromOptions(options) : options.Format;
-
-            switch (format)
+            // TODO: transfer all headers instead of a few?
```

Comment:
TODO: headers

Reply:
replaced by https://github.com/annelo-msft/openai-in-typespec/pull/1/files#r1529088375